Yoast/wordpress-seo
# Add a "Configuration required" overlay to premium cards when Conf. workout is not finished

## Context


* We want the configuration workout to be a pre-requisite for the Cornerstone and Orphaned content workouts.

## Summary


This PR can be summarized in the following changelog entry:

* Requires the Configuration workout to be completed to access the Cornerstone and Orphaned Content workouts.

## Relevant technical choices:

*

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Install and activate Premium 17.7
* visit the workouts page
* if the configuration workout is not completed, you should see a "Configuration required" overlay as per [design](https://www.sketch.com/s/02ece874-aaed-4922-bae7-f5ae46a0f94c/a/rbjlAZD/play) on the other two workout cards
* if so, complete the configuration workout, then go back to the workouts overview (without reloading)
  * the other two workouts should now display the usual buttons to start them.
* re-start the conf. workout, then go back to the workouts overview before finishing it
  * you should see again a "Configuration required" overlay as per [design](https://www.sketch.com/s/02ece874-aaed-4922-bae7-f5ae46a0f94c/a/rbjlAZD/play) on the other two workout cards
* disable Premium
* visit the workouts page
* you should always see the upsell buttons regardless of the state of the conf. workout.

_Test Editor_
* Install and activate Free and Premium.
* Log in as an editor and navigate to the workouts page.
* You should NOT see the configuration workout, but you should see the two Premium workouts (cornerstone, orphaned), which should NOT be blocked (you should be able to do both of them).

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes [DUPP-17]


[DUPP-17]: https://yoast.atlassian.net/browse/DUPP-17?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
